lib/bindConnectors/index.js

```diff
@@ -26,24 +26,38 @@ module.exports = function (config, options) {
 		// Dont wait for event loop to be empty when callback is called
 		context.callbackWaitsForEmptyEventLoop = false;
 
+
+		/*
+			If context.getRemainingTimeInMillis is available, set it's value
+			minus 10 seconds, and make sure it is positive; else set the
+			timeout to 0 (such that it is falsy)
+		*/
+		var timeRemaining = (
+			_.isPlainObject(context) && _.isFunction(context.getRemainingTimeInMillis) ?
+			context.getRemainingTimeInMillis() :
+			0
+		);
+		var promiseTimeout = timeRemaining - 10000;
+		promiseTimeout = ( promiseTimeout < 1 ? 0 : promiseTimeout );
```

Comment:
if getRemainingTimeInMillis is undefined or not a function, maybe better to log a warning and not use this promiseTimeout  behaviour at all - set promiseTimeout to undefined (as you don't really know then how long you should wait) . but it should not really happen (freeman does not use this path of code, right?)

Reply:
So, we are assuming `context.getRemainingTimeInMillis` is available when the connector is run on lambda - hence if the connector is run on anything else, we should not use a timeout. Hence I'm setting `promiseTimeout` to 0 which is falsy, since I first check for that on line 116.

So yh, anything but lambda will not have a timeout set on the promise.

Reply:
As for logging a warning, would that not assume all connectors are run on lambda? I don't think `getRemainingTimeInMillis` should be a must have or default.

Reply:
oh right, i misread the code